ucp/core.py

```diff
@@ -230,6 +235,13 @@ def __init__(self, config_dict={}, blocking_progress_mode=None):
         else:
             self.blocking_progress_mode = True
 
+        if "UCXPY_CONNECT_TIMEOUT" in os.environ:
+            self.connect_timeout = float(os.environ["UCXPY_CONNECT_TIMEOUT"])
+        elif connect_timeout is not None:
+            self.connect_timeout = connect_timeout
+        else:
+            self.connect_timeout = 5.0
+
```

Comment:
`connect_timeout` takes precedence right?
```suggestion
        if connect_timeout is None:
            self.connect_timeout = float(os.get("UCXPY_CONNECT_TIMEOUT", 5))           
        else:
            self.connect_timeout = connect_timeout
```

Reply:
I am not sure, I think a non-default `connect_timeout` argument should take precedence. Then environment variable and finally the default value.

At least, I would find it surprising if the environment variable takes precedence over an explicit argument.

Reply:
Alternatively, I have seen projects using an extra _forcing_ environment variable like: `UCXPY_CONNECT_TIMEOUT_FORCE`, which would override everything else.

Reply:
Ok, I think the force/override may be a good alternative. That is not super critical right now though and I can't spend too much time on it at the moment, so I'll just apply your suggestion and revisit this idea later.